contribute/hacker-guide.md

```diff
@@ -178,31 +156,24 @@ Now, implement your bugfix or new feature!
+  just that trait; it might also be necessary to recompile its users. The `sbt` tool is not smart enough to do that, which might lead to
+  very strange errors. Full rebuilds fix the problem. Fortunately that's rarely necessary, because full rebuilds take a lot of time -- the same 8-30 minutes as mentioned above.
 * Even on solid state drives packaging Scala distribution (i.e. creating jars from class files) is a non-trivial task. To save time here,
-  some people in our team do `ant quick.comp` instead of `ant` and then create custom scripts ([here](https://github.com/adriaanm/binfu/blob/master/scafu.sh) are some examples to get you started) to launch Scala from `build/quick/classes`.
+  some people in our team do `sbt compile` instead of `sbt dist/mkPack` and then create custom scripts using `sbt/mkBin` to launch Scala from `./build/quick/bin`. Also see [the Scala README](https://github.com/scala/scala#incremental-compilation) for tips on speeding up compile times.
 * Don't underestimate the power of `print`. When starting with Scala, I spent a lot of time in the debugger trying to figure out how
   things work. However later I found out that print-based debugging is often more effective than jumping around. While it might be obvious
-  to some, I'd like to explicitly mention that it's also useful to print stack traces to understand the flow of execution. When working with `Trees`, you might want to use `showRaw` to get the `AST` representation.
-* You can publish your newly-built scala version locally to use it from sbt. Here's how:
+  to some, I'd like to explicitly mention that it's also useful to print stack traces to understand the flow of execution. When working with `Trees`, you might want to use the Scala  `showRaw` to get the `AST` representation.
+* You can publish your newly-built scala version locally using `sbt publishLocal` to use it from `sbt`. Here's how:
 
-         $ ant publish-local-opt -Dmaven.version.suffix="-test"
+         $ sbt publishLocal // This may take a while
+         ...
          $ sbt
-         [info] Set current project to test (in build file:/Users/georgii/workspace/test/)
-         > set resolvers += Resolver.mavenLocal
-         [info] Defining *:resolvers
-         [info] The new value will be used by *:externalResolvers
-         [info] Reapplying settings...
-         [info] Set current project to test (in build file:/Users/georgii/workspace/test/)
-         > ++2.12.0-test
-         [info] Setting version to 2.12.0-test
-         [info] Set current project to test (in build file:/Users/georgii/workspace/test/)
+         ...
          > console
          [info] Starting scala interpreter...
-         [info]
-         Welcome to Scala version 2.12.0-20140623-155543-8bdacad317 (Java HotSpot(TM) 64-Bit Server VM, Java 1.7.0_51).
+         [info] 
+         Welcome to Scala version 2.10.6 (OpenJDK 64-Bit Server VM, Java 1.8.0_91).
```

Comment:
The difference when passing `-Dstarr.version=2.12.0-SNAPSHOT` is the compiler that is used to compile the Scala project (library, compiler, scaladoc tool, etc). By default our sbt build uses the scala version defined in `versions.properties` (https://github.com/scala/scala/blob/2.12.x/versions.properties#L11).

In most cases this does not matter, only if you make changes to the code generator, then those will be reflected in the library's bytecode only after a bootstrap.

I think I see now what confused you: running `sbt console` doesn't give you a REPL of a local scala version. In fact, in the current sbt build, it gives you a Scala 2.10.6 REPL.

If you want to run a REPL built from the sources of your Scala checkout, you can run `sbt scala`, see https://github.com/scala/scala#using-the-sbt-build. Or you can use the bash scripts in `build/quick/bin` generated by `dist/mkBin`.